src/Microsoft.Restier.Publishers.OData/Model/RestierModelBuilder.cs

```diff
@@ -49,7 +55,19 @@ class RestierModelBuilder : IModelBuilder
 
             context.EntitySetTypeMapCollection.Clear();
 
+            ExtendModel(builder);
+
             return builder.GetEdmModel();
         }
+
+        /// <summary>
+        /// This method allow user to extend the model with more entity type and entity set before operation model builder starts
+        /// A sub class need to be implemented and register as DI service like "AddTransient&lt;RestierModelBuilder, CustomizedRestierModelBuilder&gt;"
+        /// </summary>
+        /// <param name="builder"></param>
+        public virtual void ExtendModel(ODataConventionModelBuilder builder)
```

Comment:
As we have DI now, such virtual methods could be abstracted into an interface, like IModelExtender, so that consumer won't have to inherit RestierModelBuilder. And that way even chained model extenders are automatically supported.